src/modules/mana/sagas.spec.ts

```diff
@@ -0,0 +1,263 @@
+    '00000000000000000000000000000000000000',
+  polygonWithdrawalInput:
+    '0x2e1a7d4d0000000000000000000000000000000000000000000000000de0b6b3a7640000',
+  sendInput:
+    '0x07bd3522000000000000000000000000882da5967c435ea5cc6b09150d55e8304b838f45' +
+    '00000000000000000000000000000000000000000000000000000000000000400000000000' +
+    '0000000000000000000000000000000000000000000000000001240c53c51c000000000000' +
+    '0000000000002f89ed84e0413960d9adf8d57dd56c2c2f5076cc0000000000000000000000' +
+    '0000000000000000000000000000000000000000a0209b17f3412e6f38c3eec2ca188d67f4' +
+    '40ce21058b00b5fbf05ac9b5bd7cfe435e267d80979247ad077bb80a9fb00d2f206caf8c02' +
+    '192d5cc69791195cc13c3f0000000000000000000000000000000000000000000000000000' +
+    '00000000001b00000000000000000000000000000000000000000000000000000000000000' +
+    '44a9059cbb0000000000000000000000002f89ed84e0413960d9adf8d57dd56c2c2f5076cc' +
+    '0000000000000000000000000000000000000000000000000de0b6b3a76400000000000000' +
+    '00000000000000000000000000000000000000000000000000000000000000000000000000' +
+    '0000000000000000000000000000',
+}
+
+describe('handleImportWithdrawalRequest', () => {
+  const handleTest = ({
+    txHash,
+    expectedActions,
+    address,
+    sendResponse,
+    isERC20ExitProcessed,
+  }: {
+    txHash: string
+    expectedActions: any[]
+    address?: string
+    sendResponse?: { input: string; from: string }
+    isERC20ExitProcessed?: boolean
+  }) => {
+    const { chainId, timestamp, network } = data
+
+    const maticPOSClient = {
+      isERC20ExitProcessed: jest.fn().mockResolvedValue(isERC20ExitProcessed),
+    }
+
+    const networkProvider = {
+      send: jest.fn().mockResolvedValue(sendResponse),
+    }
+
+    let test = expectSaga(
+      handleImportWithdrawalRequest,
+      importWithdrawalRequest(txHash)
+    ).provide([
+      [select(getAddress), address],
+      [call(getChainIdByNetwork, network), chainId],
+      [call(getTimestamp), timestamp],
+      [call(getMaticPOSClient), maticPOSClient],
+      [call(getNetworkProvider, chainId), networkProvider],
+    ])
+
+    expectedActions.forEach((ea) => (test = test.put(ea)))
+
+    return test.run()
+  }
+
+  describe('given valid data', () => {
+    const getExpectedActions = () => {
+      const { address, txHash, timestamp, chainId, amount } = data
+
+      return [
+        importWithdrawalSuccess(),
+        fetchTransactionRequest(
+          address,
+          txHash,
+          initiateWithdrawalSuccess(amount, chainId, txHash)
+        ),
+        watchWithdrawalStatusSuccess({
+          amount,
+          initializeHash: txHash,
+          status: WithdrawalStatus.PENDING,
+          finalizeHash: null,
+          from: address,
+          timestamp,
+        }),
+      ]
+    }
+
+    describe('when data is for a meta transaction', () => {
+      it('should dispatch desired actions', () => {
```

Comment:
When writing the description of the expectations it's important to at least name what's expected. For example:
`should dispatch the actions to, signal the withdrawal success, the fetch of the transaction, and the watch for the transaction status`. It is a bit long, but at least we know what is actually expected.
Knowing what is expected from the test is important to check if the test does what we think it does and to check if given the context, the path of code that we want to exercise is exercised.